sql/core/src/main/scala/org/apache/spark/sql/execution/limit.scala

```diff
@@ -38,12 +38,18 @@ case class CollectLimitExec(limit: Int, child: SparkPlan) extends UnaryExecNode
   override def outputPartitioning: Partitioning = SinglePartition
   override def executeCollect(): Array[InternalRow] = child.executeTake(limit)
   private val serializer: Serializer = new UnsafeRowSerializer(child.output.size)
-  override lazy val metrics = SQLShuffleMetricsReporter.createShuffleReadMetrics(sparkContext)
+  private val writeMetrics = SQLShuffleWriteMetricsReporter.createShuffleWriteMetrics(sparkContext)
+  override lazy val metrics =
```

Comment:
this is somewhat confusing. I'd create a variable for the read metrics so you can pass just that into the ShuffledRDD.

Reply:
Thanks, make sense, I'll change to separate both read/write metrics and pass them.